manifests/init.pp

```diff
@@ -57,29 +57,29 @@
 # @author Ger Apeldoorn <info@gerapeldoorn.nl>
 #
 class rhsm (
-  Optional[String[1]]    $rh_user               = undef,
-  Optional[String[1]]    $rh_password           = undef,
-  Optional[String[1]]    $org                   = undef,
-  Optional[String[1]]    $activationkey         = undef,
-  Optional[Stdlib::Fqdn] $proxy_hostname        = undef,
-  Optional[String[1]]    $proxy_scheme          = 'http',
-  Optional[Stdlib::Port] $proxy_port            = undef,
-  Optional[String[1]]    $proxy_user            = undef,
-  Optional[String[1]]    $proxy_password        = undef,
-  Stdlib::Httpurl        $baseurl               = 'https://cdn.redhat.com',
-  Stdlib::Fqdn           $servername            = 'subscription.rhsm.redhat.com',
-  Stdlib::Absolutepath   $serverprefix          = '/subscription',
-  Stdlib::Port           $serverport            = 443,
-  Stdlib::Absolutepath   $ca_cert_dir           = '/etc/rhsm/ca/',
-  String[1]              $repo_ca_cert_filename = 'redhat-uep.pem',
-  Optional[String[1]]    $repo_ca_cert_source   = undef,
-  Integer[0,1]           $manage_repos          = 1,
-  Integer[0,1]           $full_refresh_on_yum   = 0,
-  String[1]              $package_ensure        = 'installed',
-  Array[String[1]]       $enabled_repo_ids      = [],
-  Integer[0,1]           $inotify               = 1,
-  Integer[0]             $server_timeout        = 180,
-  Integer[0]             $process_timeout       = 300,
+  Optional[String[1]]             $rh_user               = undef,
+  Optional[String[1]]             $rh_password           = undef,
+  Optional[String[1]]             $org                   = undef,
+  Optional[String[1]]             $activationkey         = undef,
+  Optional[Stdlib::Fqdn]          $proxy_hostname        = undef,
+  Optional[Enum['http', 'https']] $proxy_scheme          = 'http',
```

Comment:
There's no point in making this parameter `Optional` (which means it can be `undef` in addition to `http` or `https`) since you're providing a default to  `http`. 
```suggestion
  Enum['http', 'https'] $proxy_scheme          = 'http',
```

Reply:
@raphink I'm planning on transferring https://rubygems.org/gems/puppet-lint-optional_default-check to Vox if there's enough interest.